pyonfx/timestamps.py

```diff
@@ -0,0 +1,681 @@
+        """
+
+        if os.path.isfile(path_to_timestamps_file_or_content):
+            with open(path_to_timestamps_file_or_content, "r") as f:
+                timestamps, last_frame_time, fpms = TimestampsFileParser.parse_file(
+                    f, rounding_method
+                )
+        else:
+            f = StringIO(path_to_timestamps_file_or_content)
+            timestamps, last_frame_time, fpms = TimestampsFileParser.parse_file(
+                f, rounding_method
+            )
+
+        return cls(
+            rounding_method=rounding_method,
+            timestamps=timestamps,
+            normalize=normalize,
+            fpms=fpms,
+            last_frame_time=last_frame_time,
+        )
+
+    @classmethod
+    def _from_ffprobe(
+        cls: Timestamps,
+        video_path: os.PathLike[str],
+        index: int,
+        normalize: bool,
+        rounding_method: RoundingMethod,
+    ) -> Timestamps:
+        """Create timestamps based on the ``video_path`` provided.
+
+        Note:
+            This method requires the ``ffprobe`` program to be available.
+
+        Parameters:
+            video_path (os.PathLike[str]): A Video path.
+            index (int, optional): Stream index of the video.
+            normalize (bool, optional): If True, it will shift the timestamps to make them start from 0. If false, the option does nothing.
+            rounding_method (RoundingMethod, optional): A rounding method. See the comment in Timestamps description about floor vs round.
+        Returns:
+            An Timestamps instance.
+        """
+
+        def get_timestamps(packets) -> Tuple[Fraction, Fraction, List[int]]:
+            timestamps: List[int] = []
+
+            lowest_timestamp: Fraction = None
+            highest_timestamp: Fraction = None
+
+            for packet in packets:
+                timestamp = Fraction(
+                    packet.get("pts_time", packet.get("dts_time"))
+                ) * Fraction(1000)
+                if highest_timestamp is None or highest_timestamp < timestamp:
+                    highest_timestamp = timestamp
+                if lowest_timestamp is None or lowest_timestamp > timestamp:
+                    lowest_timestamp = timestamp
+                timestamps.append(rounding_method(timestamp))
+
+            timestamps.sort()
+            return lowest_timestamp, highest_timestamp, timestamps
+
+        # Verify if ffprobe is installed
+        if shutil.which("ffprobe") is None:
+            raise Exception("ffprobe is not in the environment variable.")
+
+        cmd = [
+            "ffprobe",
+            "-select_streams",
+            f"{index}",
+            "-show_entries",
+            "packet=pts_time,dts_time:stream=codec_type,time_base:format=format_name",
+            f"{video_path}",
+            "-print_format",
+            "json",
+        ]
+        ffprobe_output = subprocess.run(cmd, capture_output=True, text=True)
```

Comment:
ffprobe seems to always output the result in utf-8 has said [here](https://trac.ffmpeg.org/ticket/8890#comment:5).
```suggestion
        ffprobe_output = subprocess.run(cmd, capture_output=True, text=True, encoding="utf-8")
```